app/javascript/components/explore/StudyGeneField.jsx

```diff
@@ -30,14 +30,29 @@ export default function StudyGeneField({ genes, searchGenes, allGenes, speciesLi
     * an array of already entered genes (geneArray),
     * and the current text the user is typing (inputText) */
   const [geneArray, setGeneArray] = useState(enteredGeneArray)
-
   const [showEmptySearchModal, setShowEmptySearchModal] = useState(false)
 
+  const [badGeneSet, setBadGeneSet] = useState(new Set([]))
+  const [showBadGeneChoice, setShowBadGeneChoice] = useState(false)
+
   /** handles a user submitting a gene search */
   function handleSearch(event) {
     event.preventDefault()
     const newGeneArray = syncGeneArrayToInputText()
-    if (newGeneArray && newGeneArray.length) {
+
+    let shouldSearch = true
+    if (newGeneArray) {
+      newGeneArray.forEach(gene => {
+        // if an entered gene is not in the valid gene options for the study
+        if (geneOptions.length > 0 && !geneOptions.find(geneOpt => geneOpt.label === gene.label) && !badGeneSet?.has(gene.label)) {
```

Comment:
BLOCKER: how does this handle case sensitivity?  We don't require the user to have the correct casing (e.g. they can type 'pTen' and it will match 'Pten'), but this looks like it's requiring exact match.

Reply:
oo interesting, good catch, the autofill kept fixing the casing when I was testing so I didn't think about that - will update this